src/components/external/CapitalOverview/components/GrantAction/GrantActionContainer.tsx

```diff
@@ -0,0 +1,176 @@
+import { GRANT_ACTION_CLASS_NAMES } from './constants';
+import './GrantAction.scss';
+import Alert from '../../../../internal/Alert/Alert';
+import { AlertTypeOption } from '../../../../internal/Alert/types';
+import Button from '../../../../internal/Button';
+import { useConfigContext } from '../../../../../core/ConfigContext';
+import { EMPTY_OBJECT } from '../../../../../utils';
+import { getTopWindowHref, setTopWindowHref } from './utils';
+import { ButtonVariant } from '../../../../internal/Button/types';
+import { IGrant } from '../../../../../types';
+import useMutation from '../../../../../hooks/useMutation/useMutation';
+
+type ActionType = NonNullable<IGrant['missingActions']>[number]['type'];
+
+export const GrantActionContainer: FunctionalComponent<{ missingActions: IGrant['missingActions']; offerExpiresAt?: string; className?: string }> = ({
+    missingActions = [],
+    offerExpiresAt,
+    className,
+}) => {
+    const { i18n, updateCore } = useCoreContext();
+    const { dateFormat } = useTimezoneAwareDateFormatting();
+    const { endpoints } = useConfigContext();
+
+    const ACTION_CONFIG = useMemo(
+        () =>
+            ({
+                signToS: {
+                    getTitle: (formattedDate: string | undefined) =>
+                        `${i18n.get('capital.signTermsAndConditionsToReceiveFunds')}${
+                            formattedDate ? ` ${i18n.get('capital.thisOfferExpiresOn', { values: { date: formattedDate } })}` : ''
+                        }`,
+                    buttonLabelKey: 'capital.goToTermsAndConditions',
+                },
+                AnaCredit: {
+                    getTitle: (formattedDate: string | undefined) =>
+                        i18n.get(
+                            formattedDate
+                                ? 'capital.weNeedABitMoreInformationToProcessYourFundsPleaseCompleteThisActionBy'
+                                : 'capital.weNeedABitMoreInformationToProcessYourFundsPleaseCompleteThisAction',
+                            { values: formattedDate ? { date: formattedDate } : undefined }
+                        ),
+                    buttonLabelKey: 'capital.submitInformation',
+                },
+            } as const),
+        [i18n]
+    );
+
+    // Use local state to track which action is loading
+    const [loadingAction, setLoadingAction] = useState<ActionType | null>(null);
+
+    const onRedirect = useCallback((data: { url: string } | undefined) => {
+        if (data?.url) {
+            setTopWindowHref(data.url);
+        } else {
+            // If the request was successful but no URL was returned, reset loading state.
+            setLoadingAction(null);
+        }
+    }, []);
+
+    const actionMutation = useMutation({
+        queryFn: (actionType: ActionType) => {
+            const endpoint = actionType === 'signToS' ? endpoints.signToSActionDetails : endpoints.anaCreditActionDetails;
```

Comment:
I think we should be more explicit with anaCredit action as shown below.
```suggestion
            const endpoint = actionType === 'signToS'
              ? endpoints.signToSActionDetails
              : actionType === 'anaCredit'
              ? endpoints.anaCreditActionDetails
              : null;
```

Why?
Consider this case, we introduce a new action. The backend eventually returns this new action but the SDK is not updated and the component does not handle this new action. In this case, the component will unintentionally handle this new action as it was anaCredit.

Reply:
✅